deltachat-ios/DC/DcAccount.swift

```diff
@@ -45,6 +45,23 @@ public class DcAccounts {
         return DcUtils.copyAndFreeArray(inputArray: cAccounts)
     }
 
+    public func getAllSorted() -> [Int] {
+        return getAll().sorted { a, b in
+            // no need to check for equality as sorted() is guaranteed to be stable;
+            // meaning it preserves the relative order of elements that compare as equal here
+            let orderA = get(id: a).getConfigInt("ui.ios.account_order"), orderB = get(id: b).getConfigInt("ui.ios.account_order")
+            return orderA > orderB
+        }
+    }
+
+    public func moveToTop(id: Int) {
+        var maxOrder = 0
+        getAll().forEach {
+            maxOrder = max(maxOrder, get(id: $0).getConfigInt("ui.ios.account_order"))
+        }
```

Comment:
what about something like this:

```swift
let highestIndex = getAll()
    .compactMap { get(id: $0).getConfigInt("ui.ios.account_order") }
    .max() ?? 0
get(id: id).setConfigInt("ui.ios.account_order", highestIndex + 1)
```

instead of `forEach`?

Reply:
yeah, maybe more modern - and one line less :)